packages/salesforcedx-vscode-core/src/orgBrowser/orgMetadata.ts

```diff
@@ -0,0 +1,145 @@
+      cwd: getRootWorkspacePath()
+    }).execute(cancellationToken);
+
+    execution.processExitSubject.subscribe(async data => {
+      this.logMetric(execution.command.logName, startTime);
+      buildTypesList(this.outputPath);
+    });
+    notificationService.reportExecutionError(
+      execution.command.toString(),
+      (execution.stderrSubject as any) as Observable<Error | undefined>
+    );
+    channelService.streamCommandOutput(execution);
+    ProgressNotification.show(execution, cancellationTokenSource);
+    taskViewService.addCommandExecution(execution, cancellationTokenSource);
+  }
+}
+
+const workspaceChecker = new SfdxWorkspaceChecker();
+const parameterGatherer = new EmptyParametersGatherer();
+
+export async function forceDescribeMetadata(outputPath?: string) {
+  if (!isNullOrUndefined(outputPath)) {
+    const describeExecutor = new ForceDescribeMetadataExecutor(outputPath);
+    const commandlet = new SfdxCommandlet(
+      workspaceChecker,
+      parameterGatherer,
+      describeExecutor
+    );
+    await commandlet.run();
+  }
+}
+
+async function getMetadataTypesPath(): Promise<string | undefined> {
+  if (hasRootWorkspace()) {
+    const workspaceRootPath = getRootWorkspacePath();
+    const defaultUsernameOrAlias = await getDefaultUsernameOrAlias();
+    const defaultUsernameIsSet = typeof defaultUsernameOrAlias !== 'undefined';
+
+    if (defaultUsernameIsSet) {
+      const username = await OrgAuthInfo.getUsername(defaultUsernameOrAlias!);
+      const metadataTypesPath = path.join(workspaceRootPath, '.sfdx', 'orgs', username, 'metadata', 'metadataTypes.json');
+      return metadataTypesPath;
+    } else {
+      throw new Error(nls.localize('error_no_default_username'));
+    }
+  } else {
+    throw new Error(nls.localize('cannot_determine_workspace'));
+  }
+}
+
+export async function getDefaultUsernameOrAlias(): Promise<string | undefined> {
+  if (hasRootWorkspace()) {
+    return await OrgAuthInfo.getDefaultUsernameOrAlias();
+  }
+}
+
+export type MetadataObject = {
+  directoryName: string;
+  inFolder: boolean;
+  metaFile: boolean;
+  suffix: string;
+  xmlName: string;
+};
+
+export function buildTypesList(metadataTypesPath: string) {
+  if (!isNullOrUndefined(metadataTypesPath)) {
+    const fileData = JSON.parse(fs.readFileSync(metadataTypesPath, 'utf8'));
+    const metadataObjects = fileData.metadataObjects;
+    const metadataTypes = [];
+    for (const index in metadataObjects) {
+      if (!isNullOrUndefined(metadataObjects[index].xmlName)) {
+        metadataTypes.push(metadataObjects[index].xmlName);
+      }
+    }
+  } else {
+    throw new Error('There was an error retrieving metadata type information.');
+  }
+}
+
+export async function onUsernameChange() {
+  const metadataTypesPath = await getMetadataTypesPath();
+  if (!isNullOrUndefined(metadataTypesPath) && fs.existsSync(metadataTypesPath)) {
+    buildTypesList(metadataTypesPath);
```

Comment:
I don't think we need to load the data (.json file created by the cli command above) in memory every time the org credentials are changed/added to the `.sfdx-config.json`. I'd imagine this would only be the entry point to getting data from the org if it's not available locally.